### ISSUE TITLE
Buildings rendering differently on Chef instance

### ISSUE THREAD

**Issue by robhawkes** [polygon-city/chef-buildingdb#5]
Buildings are rendering different on the Chef instance compared to my local copy. I think it might have something to with assimp acting differently as 20 days ago I updated it to run some checks to fix things like broken normals (perhaps this isn't being run).

Can someone confirm the HEAD commit for the model-converter dependency being used on the Chef instance? It should be [6e6ff25b176669cd527fb1349525bbf1c779fcb5](https://github.com/polygon-city/model-converter/commit/6e6ff25b176669cd527fb1349525bbf1c779fcb5). If it's not that then I've probably just not defined the dependency correctly [in package.json](https://github.com/polygon-city/building-database/blob/master/package.json#L37) and it's not installing the latest changes.

Local instance:
![](http://cl.ly/image/22210c1a3W1P/Screen%20Shot%202015-03-27%20at%2011.46.42.png)

Chef instance:
![](http://cl.ly/image/2d143m313N1Z/Screen%20Shot%202015-03-27%20at%2011.46.50.png)

**Comment by bcamper** [comment]
@baldur could you help here, please?

**Comment by baldur** [comment]
@robhawkes looks like it is just pointed to head 

```
"model-converter": "git://github.com/polygon-city/model-converter"
```

In the node_modules directory it's head is 

```
node_modules/model-converter$ git rev-parse HEAD
b8099838744978dd3127cc04e73be378a658510a
```

**Comment by robhawkes** [comment]
I've pushed a new version number to the `model-converter` repo so hopefully that updates properly on the next CircleCI build. Interestingly, b8099838744978dd3127cc04e73be378a658510a isn't a commit in the repo that I can find.

**Comment by robhawkes** [comment]
It looks like CircleCI is using the updated `model-converter` dependency so that's good. Once #6 is sorted out then we should always have the latest version on the dev server.

**Comment by baldur** [comment]
@robhawkes great ... closing for now then

**Comment by robhawkes** [comment]
I've re-opened for now as the buildings are still different on the dev server for some reason – both larger and lit differently. I haven't pinned it down yet but the new `/ping` endpoint shows that we're at least using the latest repo.

I'm wondering if it's the fact we're using a slightly different assimp. Do you know which version has been installed on the dev server? This is the response to `assimp version` for the OSX build:

```
Version 3.1 -debug -noboost -shared -st (GIT commit 94f6130)
```

If it's the same version then I'm out of ideas for the moment, so I'm hoping there's a difference here. It's probably worth my time to work out how to build a linux binary from the same source as the OSX one.

**Comment by baldur** [comment]
Version 3.0--shared -st  (SVNREV 1262)

I can look into trying to upgrade easily ... is there a ppa that you know
of that is more recent?

**Comment by robhawkes** [comment]
Well at least it looks like this might be the cause!

I can't find any updated packages which is why I resorted to building my own, with what seems to be the master branch at the time Polygon City started: https://github.com/assimp/assimp/commit/94f6130

Apparently there _might_ be binaries here but the link isn't working for me: http://sourceforge.net/projects/assimp/files/assimp-3.1/

I've also just found the build FAQ so if we do need to create our own binaries than I should be able to work it out (though I don't have a copy of Linux at the moment and my Internet is appalling): https://github.com/assimp/assimp/wiki/FAQ

**Comment by baldur** [comment]
ok, I'll do the same on the server

**Comment by baldur** [comment]
@robhawkes can you post a link to a file that I can use for testing

**Comment by robhawkes** [comment]
Try this one, it's my go-to for testing at the moment: http://cl.ly/2o2u1L351a15

It should show up at exactly the right scale when uploaded. Currently, the dev server shows it really really large (as with all the other uploads).

**Comment by baldur** [comment]
@robhawkes can you try it now and see if it is doing the correct thing. I installed things manually so I want to verify that this was the issue before I package up the work

**Comment by robhawkes** [comment]
Yeah, that worked! Seems to be the same scale now as my local setup and also the lighting on the models is correct. What version did you end up using?

**Comment by baldur** [comment]
@robhawkes just installed from head ;)
Version 3.1 -debug -noboost -shared -st (GIT commit 45cc0b5)

**Comment by robhawkes** [comment]
It seems to be working great now, so happy to go with that version. Did you package it into a Linux binary, or install it some other way? If so, it'd be good to add it to the [`model-converter`](https://github.com/polygon-city/model-converter) repo, or send it to me and I'll add it.

**Comment by baldur** [comment]
@robhawkes I don't know assimp well enough to know how realistic it is to bundle it with model-converter we can how ever document the process of installing it outside the chef context. I'll write it here not sure where you wanted to include it in the model-converter repo.

```
git clone https://github.com/assimp/assimp.git
cd assimp
cmake -G 'Unix Makefiles'
sudo make install
```

Then /usr/local/lib has to be added to the library path which can be done in a couple of ways 

```
export LD_LIBRARY_PATH=/usr/local/lib
```

or more permanently by adding a file such as

```
sudo echo "/usr/local/lib" > /etc/ld.so.conf.d/assimp.conf
sudo ldconfig
```

**Comment by baldur** [comment]
@robhawkes does npm offer the ability to --compile from source? that might be an option for managing this dependency in model-converter. I'd be happy to help you figure it out.

**Comment by robhawkes** [comment]
There's [`node-gyp`](https://github.com/TooTallNate/node-gyp) which could work for compilation, though I've not used it before so unsure how complex it makes things. I assume it allows you to set the various paths and stuff so the compiled module can actually be used.
